### PR TITLE
Converted BurnState Visualizer

### PR DESCRIPTION
**Pull request by Bixkitts**
Part of #8995

### CODE REVIEW

File: Content.Client/Smoking/BurnStateVisualizerSystem.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
+using Robust.Client.GameObjects;
+using Content.Shared.Smoking;
+
+namespace Content.Client.Smoking;
+
+public sealed class BurnStateVisualizerSystem : VisualizerSystem<BurnStateVisualsComponent>
+{
+
+    [Dependency] private readonly IEntityManager _entMan = default!;
```
Comment (vulppine) [inline Content.Client/Smoking/BurnStateVisualizerSystem.cs:9]: don't need this, EntitySystems have this integrated and have helper methods already

File: Content.Client/Smoking/BurnStateVisualizerSystem.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    private void OnComponentInit(EntityUid uid, BurnStateVisualsComponent component, ComponentInit args)
+    {
+        IoCManager.InjectDependencies(this);
+    }
+
+    protected override void OnAppearanceChange(EntityUid uid, BurnStateVisualsComponent component, ref AppearanceChangeEvent args)
+    {
+
+        if (!_entMan.TryGetComponent(uid, out SpriteComponent? sprite))
```
Comment (vulppine) [inline Content.Client/Smoking/BurnStateVisualizerSystem.cs:26]: AppearanceChangeEvent already has this, you can just check if that's null or not

File: Content.Client/Smoking/BurnStateVisualizerSystem.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+            return;
+
+        if (!_appearance.TryGetData<SmokableState>(uid, SmokingVisuals.Smoking, out var burnState))
```
Comment (vulppine) [inline Content.Client/Smoking/BurnStateVisualizerSystem.cs:29]: AppearanceChangeEvent already has the dictionary that contains all the appearance data for the given entity, you can just check that instead

File: Content.Client/Smoking/BurnStateVisualizerSystem.cs [hunk 4]
```diff
@@ -0,0 +1,42 @@
+using Robust.Client.GameObjects;
+using Content.Shared.Smoking;
+
+namespace Content.Client.Smoking;
+
+public sealed class BurnStateVisualizerSystem : VisualizerSystem<BurnStateVisualsComponent>
+{
+
+    [Dependency] private readonly IEntityManager _entMan = default!;
+    [Dependency] private readonly SharedAppearanceSystem _appearance = default!;
```
Comment (vulppine) [inline Content.Client/Smoking/BurnStateVisualizerSystem.cs:10]: you very likely will not need this dependency

File: Content.Client/Smoking/BurnStateVisualizerSystem.cs [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<BurnStateVisualsComponent, ComponentInit>(OnComponentInit);
+    }
+
+    private void OnComponentInit(EntityUid uid, BurnStateVisualsComponent component, ComponentInit args)
+    {
+        IoCManager.InjectDependencies(this);
```
Comment (vulppine) [inline Content.Client/Smoking/BurnStateVisualizerSystem.cs:20]: definitely don't need this at all, entity system type classes already do this for you

**Review by vulppine — CHANGES_REQUESTED** [review]

File: Content.Client/Smoking/BurnStateVisualizerSystem.cs (hunk 1 above)
Comment (Bixkitts) [reply Content.Client/Smoking/BurnStateVisualizerSystem.cs]: removed

File: Content.Client/Smoking/BurnStateVisualizerSystem.cs (hunk 2 above)
Comment (Bixkitts) [reply Content.Client/Smoking/BurnStateVisualizerSystem.cs]: Checked

File: Content.Client/Smoking/BurnStateVisualizerSystem.cs (hunk 3 above)
Comment (Bixkitts) [reply Content.Client/Smoking/BurnStateVisualizerSystem.cs]: Checked dictionary for appearance data

File: Content.Client/Smoking/BurnStateVisualizerSystem.cs (hunk 4 above)
Comment (Bixkitts) [reply Content.Client/Smoking/BurnStateVisualizerSystem.cs]: Removed dependencies

File: Content.Client/Smoking/BurnStateVisualizerSystem.cs (hunk 5 above)
Comment (Bixkitts) [reply Content.Client/Smoking/BurnStateVisualizerSystem.cs]: Erased

**Comment by Bixkitts** [comment]
First pull request and C# endeavour ever, thank you for the comments this simple refactor is helpful for figuring out what's going on.
I'll do a couple more of these soon as this one's correct!

File: Content.Client/Smoking/BurnStateVisualsComponent.cs [hunk 6]
```diff
@@ -0,0 +1,13 @@
+namespace Content.Client.Smoking;
+
+    [RegisterComponent]
+    public sealed class BurnStateVisualsComponent : Component
+    {
+        [DataField("burntIcon")]
+        public string _burntIcon = "burnt-icon";
```
Comment (vulppine) [inline Content.Client/Smoking/BurnStateVisualsComponent.cs:7]: turn these from the underscore names to something like “BurntIcon”, the underscore names are meant for private fields but if they’re public they need a naming scheme similar to other public classes

**Review by vulppine — CHANGES_REQUESTED** [review]

**Review by AJCM-git — CHANGES_REQUESTED** [review]
Build errors, there is something weird with your formater as well, BurnStateVisualsComponent.cs has an unnecessary extra level of indentation  for the class and we normally put an space after the if and before the parenthesis `if (  `

**Comment by Bixkitts** [comment]
Fixed, oversight with naming and indentation.
( DataField("camelCase"), public var PascalCase. )
Still acclimatizing.
Will make sure it tests successful in future cases before review.
Going to look into GenericVisualizerSystem before next review.

**Review by vulppine — APPROVED** [review]

**Pull request merged**